### ISSUE TITLE
S1450 false positive on unused field

### ISSUE THREAD

**Issue by sm-g** [SonarSource/sonar-csharp#1818]
```cs
public class Foo {
   private readonly ILogger _logger;

   public Foo(ILogger logger)
   {
       _logger = logger;
   }
}
```

7.4.0.6452
VS 15.8.1

**Comment by Evangelink** [comment]
Hi @sm-g,

I am not sure what to say in such case because obviously if this is the only place where `_logger` is used in this class I do think this is unused field and that this needs to be removed.

Could you let me know if you are using the field somewhere else in the class?

Cheers,
Amaury

**Comment by sm-g** [comment]
No, field not used.

You probably have changed the Action for this rule and I thought that it is new bug (before update of nuget this code was fine).

**Comment by Evangelink** [comment]
We rewrote the rule to reduce FPs and improve detection of unused members :)
